gst/datarepo/gstdatareposink.c

```diff
@@ -764,6 +719,29 @@ gst_data_repo_sink_write_json_meta_file (GstDataRepoSink * sink)
   return ret;
 }
 
+/**
+ * @brief Check if the tensors are static or not
+ */
+static void
+gst_data_repo_sink_is_static_tensors (GstDataRepoSink * sink)
+{
+  GstStructure *structure;
+  GstTensorsConfig config;
+  tensor_format ret = _NNS_TENSOR_FORMAT_END;
+
+  g_return_if_fail (sink != NULL);
+  g_return_if_fail (sink->fixed_caps != NULL);
+
+  if (sink->data_type != GST_DATA_REPO_DATA_TENSOR)
+    return;
+
+  structure = gst_caps_get_structure (sink->fixed_caps, 0);
+  if (gst_tensors_config_from_structure (&config, structure)) {
```

Comment:
I suggest adding this into gst_data_repo_sink_set_caps() function, not state change callback.
```
gst_data_repo_sink_set_caps (GstBaseSink * bsink, GstCaps * caps)
{
...
  sink->fixed_caps = gst_caps_copy (caps);
  structure = gst_caps_get_structure (sink->fixed_caps, 0);

  if (gst_tensors_config_from_structure (&config, structure)) {
    sink->is_static_tensors = gst_tensors_config_is_static (&config);
    gst_tensors_config_free (&config);
  }
...
}
```

Reply:
Thank you for the review :+1: 
You're right too, moving the function is a good idea too.

Reply:
When I use the function I created, if function goes to set_caps, the value of pad is still not set, so I will keep the chage stage.
Thank you.

Reply:
when the pad caps is set and fixed? basesrc may call this callback if pad caps is changed.

Reply:
Thank you. :+1: